### DISCUSSION TITLE
Is it possible to configure CLion as an IDE for use with Emscripten

### DISCUSSION THREAD

**Discussion by ericoporto** [emscripten-core/emscripten#18801] [Q&A]
I use CLion a lot for my software development in C++ and maybe I should be asking in their forums, but just in case someone else uses it too here, is there any known way to get CLion configured so that it works with Emscripten?

CLion uses CMake based projects by default, and it has a powerful indexing feature, clang tidy suggestions and I also leverage additional plugins like SonarLint in it. It would be nice to have all this working in Emscripten - currently I edit things in a Linux port and then after sketching there copy and paste to the real thing I am trying to do in the actual Emscripten files. 

I have debugging in Chrome with source and everything working, but disconnected from CLion, debugging would be nice too, but it's not needed at this time, I am more interested in just being able to edit the code for emscripten with all the preprocessor and related things working in the indexing.

Oh, and I am more interested in getting it working on CLion on Windows.

**Comment by sbc100** [comment]
I don't know of any folks who are doing this, but it should be simple enough.   I imagine all your would need to do is tell CLion to run cmake with `-DCMAKE_TOOLCHAIN_FILE=/path/to/emscripten/cmake/Modules/Platform/Emscripten.cmake` (which is basically all the emcmake.py does.
